cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -0,0 +1,478 @@
+                        result.fail(new ConnectRestException(response, "Unexpected status code"));
+                    }
+                })
+                .exceptionHandler(result::fail)
+                .setFollowRedirects(true)
+                .putHeader("Accept", "application/json")
+                .end();
+        return result.future();
+    }
+
+    private Future<Void> updateLoggers(String host, int port, String desiredLogging, Map<String, Map<String, String>> fetchedLoggers) {
+        Map<String, String> updateLoggers = new LinkedHashMap<>();
+        fetchedLoggers.entrySet().forEach(entry -> {
+            // set all logger levels to OFF
+            updateLoggers.put(entry.getKey(), "OFF");
+        });
+
+        OrderedProperties ops = new OrderedProperties();
+        ops.addStringPairs(desiredLogging);
+        ops.asMap().entrySet().forEach(entry -> {
+            // set desired loggers to desired levels
+            if (entry.getKey().equals("log4j.rootLogger")) {
+                updateLoggers.put("root", entry.getValue());
+            } else if (!entry.getKey().startsWith("log4j.appender")) {
+                updateLoggers.put(entry.getKey().replace("log4j.logger.", ""), entry.getValue());
+            }
+        });
+
+        LinkedHashMap<String, String> updateSortedLoggers = sortLoggers(updateLoggers);
+        Future<Void> result = Future.succeededFuture();
+        for (Map.Entry<String, String> logger : updateSortedLoggers.entrySet()) {
+            result = result.compose(previous -> updateConnectorLogger(host, port, logger.getKey(), logger.getValue().split(",")[0].replaceAll("\\s", "")));
+        }
+        return result;
+    }
+
+    @Override
+    public Future<Void> updateConnectLoggers(String host, int port, String desiredLogging) {
+        return listConnectorLoggers(host, port).compose(fetchedLoggers -> updateLoggers(host, port, desiredLogging, fetchedLoggers));
+    }
+
+    private int numberOfDotsInString(String tested) {
+        return tested.length() - tested.replaceAll("\\.", "").length();
+    }
+
+    /**
+     * To apply loggers correctly, we need to sort them. The sorting is performed on base of logger generality.
+     * Logger "abc.company" is more general than "abc.company.name"
+     * @param loggers map of loggers to be sorted
+     * @return map of sorted loggers
+     */
+    private LinkedHashMap<String, String> sortLoggers(Map<String, String> loggers) {
+        Comparator<Map.Entry<String, String>> loggerComparator = (e1, e2) -> {
+            String k1 = e1.getKey();
+            String k2 = e2.getKey();
+            if (k1.equals("root")) {
+                // we need root logger always to be the first logger to be set via REST API
+                return Integer.MIN_VALUE;
+            }
+            if (k2.equals("root")) {
+                return Integer.MAX_VALUE;
+            }
+            return numberOfDotsInString(k1) - numberOfDotsInString(k2);
```

Comment:
Why not `k1.compareTo(k2)`? Because a prefix is < every string of which it is a prefix.